pkg/querier/iterators/chunk_merge_iterator.go

```diff
@@ -182,22 +213,27 @@ func newNonOverlappingIterator(chunks []*chunkIterator) *nonOverlappingIterator
 	}
 }
 
-func (it *nonOverlappingIterator) Seek(t int64) bool {
+func (it *nonOverlappingIterator) Seek(t int64) chunkenc.ValueType {
 	for ; it.curr < len(it.chunks); it.curr++ {
-		if it.chunks[it.curr].Seek(t) {
-			return true
+		if v := it.chunks[it.curr].Seek(t); v != chunkenc.ValNone {
+			return v
 		}
 	}
 
-	return false
+	return chunkenc.ValNone
 }
 
-func (it *nonOverlappingIterator) Next() bool {
-	for it.curr < len(it.chunks) && !it.chunks[it.curr].Next() {
+func (it *nonOverlappingIterator) Next() chunkenc.ValueType {
```

Comment:
Correct me if I'm wrong, but I think can be rewritten in a easier way:

```
for it.curr < len(it.chunks) {
  valType = it.chunks[it.curr].Next()
  if valType != chunkenc.ValNone {
    return valType
  }
}

return chunkenc.ValNone
```

Reply:
almost, adding missing `it.curr++`